modules/iasRtdProvider.js

```diff
@@ -51,94 +69,107 @@ function stringifyScreenSize() {
   return [(window.screen && window.screen.width) || -1, (window.screen && window.screen.height) || -1].join('.');
 }
 
-function getPageLevelKeywords(response) {
+function formatTargetingData(adUnit) {
   let result = {};
-  if (response.brandSafety) {
-    shallowMerge(result, response.brandSafety);
+  if (iasTargeting[BRAND_SAFETY_OBJECT_FIELD_NAME]) {
+    extend(result, iasTargeting[BRAND_SAFETY_OBJECT_FIELD_NAME]);
+  }
+  if (iasTargeting[FRAUD_FIELD_NAME]) {
+    result[FRAUD_FIELD_NAME] = iasTargeting[FRAUD_FIELD_NAME];
+  }
+  if (iasTargeting[CUSTOM_FIELD_NAME]) {
+    result[IAS_KW] = iasTargeting[CUSTOM_FIELD_NAME][IAS_KW];
+  }
+  if (iasTargeting[SLOTS_OBJECT_FIELD_NAME] && adUnit in iasTargeting[SLOTS_OBJECT_FIELD_NAME]) {
+    extend(result, iasTargeting[SLOTS_OBJECT_FIELD_NAME][adUnit]);
   }
-  result.fr = response.fr;
-  result.custom = response.custom;
   return result;
 }
 
-function shallowMerge(dest, src) {
-  utils.getKeys(src).reduce((dest, srcKey) => {
-    dest[srcKey] = src[srcKey];
-    return dest;
-  }, dest);
+function extend(dest, src) {
+  if (src) {
+    utils.getKeys(src).forEach(key => {
+      dest[key] = src[key];
+    });
+  }
+  return dest;
```

Comment:
Did you consider instead of a custom helper using the existing `utils.mergeDeep`?